src/main/java/Duke.java

```diff
@@ -1,10 +1,329 @@
+        AnchorPane.setTopAnchor(startButton, 1.0);
+        AnchorPane.setLeftAnchor(startButton, 1.0);
+
+        AnchorPane.setLeftAnchor(userInput, 2.0);
+        AnchorPane.setBottomAnchor(userInput, 1.0);
+
+        //Step 3. Add functionality to handle user input.
+        sendButton.setOnMouseClicked((event) -> {
+            handleUserInput();
+        });
+
+        userInput.setOnAction((event) -> {
+            handleUserInput();
+        });
+
+        //Scroll down to the end every time dialogContainer's height changes.
+        dialogContainer.heightProperty().addListener((observable) -> scrollPane.setVvalue(1.0));
+    }
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     *
+     * @param text String containing text to add
+     * @return a label with the specified text that has word wrap enabled.
+     */
+    private Label getDialogLabel(String text) {
+        // You will need to import `javafx.scene.control.Label`.
+        Label textToAdd = new Label(text);
+        textToAdd.setWrapText(true);
+
+        return textToAdd;
+    }
+
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        Label userText = new Label(userInput.getText());
+        Label dukeText = null;
+        try {
+            dukeText = new Label(getResponse(userInput.getText()));
+        } catch (DukeException e) {
+            e.printStackTrace();
+        }
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userText, new ImageView(user)),
+                DialogBox.getDukeDialog(dukeText, new ImageView(duke))
+        );
+        userInput.clear();
+    }
+
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    private String getResponse(String input) throws DukeException{
+        assert FILEPATH.equals("java/data/list.txt") : "wrong filepath";
+        //return "Duke heard: " + input;
+        Ui ui = new Ui(input);
+        String command = ui.getCommand();
+        String response = "";
+        if (command.equals("todo")) {
+            try {
+                //response = "todo";
+                assert isTaskListLoaded : "TaskList not loaded from storage";
+                Todo todo = new Todo(ui.getDescription());
+                tl.TL.add(todo);
+                response = ui.addedTask(todo);
+            } catch (Exception e) {
+                response = e.getMessage();
+            }
+        } else if (command.equals("deadline")) {
+            try {
+                //response = "deadline";
+                assert isTaskListLoaded : "TaskList not loaded from storage";
+                String desc = ui.getDescription();
```

Comment:
Quite a few number of if-else clauses that make it hard to follow. Consider using some form of abstraction (SLAP) or use switch statements to improve readability!

Reply:
I agree with this comment.